packages/app/src/client/utils/request.ts

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright 2021 Opstrace, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import axios from "axios";
+import * as rax from "retry-axios";
+
+const request = axios.create({
+  // For config options see https://github.com/axios/axios#request-config
+  timeout: 5000
+});
+
+request.defaults.raxConfig = {
+  retry: 3,
+  instance: request
+};
```

Comment:
Is it safe to generally just retry? Or should we only retry in certain failure scenarios?

Reply:
Depends on the specific request.

- It's the right thing to retry for all failures that result in the request not being set or the response not being received. That includes transport layer problems (network problems) and service outages.

- When the request was sent and a response was received, some responses are retryable:
    - usually _all_ 5xx ones
    - usually a 429 one, but that may be special (and require a larger delay between attempts
    - depending on the _specifics_ of the request other responses are retryable

Reply:
axios-retry probably documents what it retries by default. I suppose that it does (1) perfectly by default. I suppose it also retries all 5xx by default.

Reply:
@MoSattler I just remembered that I played with that before. See this commit for inspiration: https://github.com/opstrace/opstrace/commit/cc4464f49a3d930d6c21b7acde5cb5b658388937

Reply:
For now, I just copied the rax config you had in place there. At least as a starting point.